### PR TITLE
Add iai_callgrind benchmark for measuring memory

### PR DESCRIPTION
**Pull request by arlyon**

### CODE REVIEW

**Comment by arlyon** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/vercel/next.js/64569?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#64569** <a href="https://app.graphite.dev/github/pr/vercel/next.js/64569?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#64121** <a href="https://app.graphite.dev/github/pr/vercel/next.js/64121?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#64122](https://github.com/vercel/next.js/pull/64122) <a href="https://app.graphite.dev/github/pr/vercel/next.js/64122?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @arlyon and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 442b6bc5c52f18eed21785fe8d88acc02a909fe5

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: packages/next-swc/crates/next-build-test/benches/bench.rs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+                + Send
+                + Sync,
+        >,
+    >,
+) {
+    next_build_test::register();
+
+    let rt = tokio::runtime::Builder::new_multi_thread()
+        .enable_all()
+        .on_thread_stop(|| {
```
Comment (bgw) [inline packages/next-swc/crates/next-build-test/benches/bench.rs:82]: Can't this just be

```
.on_thread_stop(TurboMalloc::thread_stop)
```

(and shouldn't clippy detect this? maybe I'm missing something?)

File: packages/next-swc/crates/next-build-test/benches/bench.rs [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+workflow-collaboration/conformance/rules/REQUIRE_CARET_DEPENDENCIES",
+    ];
+
+    rt(Box::pin(async move {
+        let (container, _dir) = get_container();
+
+        let entrypoints = container.entrypoints().await.unwrap();
+
+        let routes = entrypoints.routes.clone().into_iter().limit(pages);
+        // .filter(|route| SELECTED_PAGES.contains(&route.0.as_str()));
```
Comment (bgw) [inline packages/next-swc/crates/next-build-test/benches/bench.rs:144]: Leftover comment?

File: packages/next-swc/crates/next-build-test/benches/bench.rs [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+    std::process::Command::new("sh")
+        .arg("-c")
+        .current_dir(tmp.path())
+        .arg(install_command)
+        .status()
+        .unwrap();
+
+    let project_path = package_path
+        .map(|pp| tmp.path().join(pp))
```
Comment (bgw) [inline packages/next-swc/crates/next-build-test/benches/bench.rs:46]: `map_or_else` should combine this with the `.unwrap_or_else` below it.

https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or_else

I thought there was a clippy lint for this?

**Review by arlyon — COMMENTED** [review]
Closing this as the overhead in recording is too much to be run in CI

**Pull request closed without merging**